sympy/solvers/diophantine.py

```diff
@@ -933,7 +933,7 @@ def _diop_quadratic(var, coeff, t):
     C = coeff[y**2]
     D = coeff[x]
     E = coeff[y]
-    F = coeff[1]
+    F = coeff[S.One]
```

Comment:
The problem is not here since 1 and `S.One` would be the same key in a dictionary. You will know you have fixed it when the test from the issue passes in your branch. You should add that test to the `test_diophantine` file, too.

Reply:
I left a suggestion on the issue page that may address the problem.

Reply:
so should i change line 936 back to original and apply your suggestion?

Reply:
Actually this line is the correct fix I think. The `coeff` variable here is `defaultdict` so this line inserts either `1` or `S.One` as a key into the dict. Probably this should not be a defaultdict so a better overall fix would be to change that. However given that it is a defaultdict the change here seems correct.

Reply:
Agree. I was focused on the value of F which does not depend on whether the key is 1 or S.One. But when 1 is not present then whatever key you use is inserted into `coeff` and the `1` is what caused problems later. So the import and change to line 1117 can be reverted.

Reply:
so just changing `coeff[1]` with `coeff[S.One]` and reverted rest of the things would fix this?